src/error_types.rs

```diff
@@ -0,0 +1,33 @@
+use std::error;
+use std::fmt;
+use std::io;
+
+#[derive(Debug, Clone)]
+pub struct GnuplotInitError;
+
+impl fmt::Display for GnuplotInitError
+{
+	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result
+	{
+		write!(
+			f,
+			"Couldn't spawn gnuplot. Make sure it is installed and available in PATH."
+		)
+	}
+}
+
+impl error::Error for GnuplotInitError
+{
+	fn source(&self) -> Option<&(dyn error::Error + 'static)>
+	{
+		None
+	}
+}
+
+impl From<io::Error> for GnuplotInitError
+{
+    fn from(_error: io::Error) -> Self
+	{
+        GnuplotInitError
```

Comment:
Would you mind sticking the io error into `GnuplotInitError` and then exposing it via the `Error::source` method (and put it into the `fmt` for good measure I guess)? I'm concerned that we've lost a little bit of debuggability here.

Reply:
I'm not sure what you're describing exactly, or at least how to accomplish it. I'm happy to make changes if you can explain it a tiny bit.

Reply:
Something like this:

```rust
#[derive(Debug)]
struct MyError
{
	inner: Box<dyn error::Error + 'static>,
}

impl From<io::Error> for MyError
{
	fn from(error: io::Error) -> Self
	{
		MyError
		{
			inner: Box::new(error),
		}
	}
}

impl fmt::Display for MyError
{
	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result
	{
		write!(
			f,
			"MyError: {}.", self.inner
		)
	}
}
```

The point being that we preserve the actual `io::Error` in case that is useful.

Reply:
That makes sense. I think it should be good now, let me know if you think it needs any more changes.